.github/workflows/check_changelog_file.yaml

```diff
@@ -0,0 +1,50 @@
+name: Check Changelog File
+
+on:
+  pull_request:
+    types: [opened]
+
+jobs:
+  # Check if the PR creates a sperate changelog file in the ".unreleased" folder
+  # This check can be disabled by adding the following line in the PR text
+  #
+  # Disable-check: force-changelog-file
+  #
+  # The change log file should have ".md" extension and if exists, check
+  # that if the file is of the expected format of bugfix/features/thanks
+  # sections.
+  checkfor-changelog-file:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Check changelog
+        run: |
+          folder=`.unreleased/features`
```

Comment:
instead of writing long scripts into yaml files - you can move it out as a separate script; it could help simplify development; faster to work with/etc...